content/actions/guides/building-and-testing-powershell.md

```diff
@@ -0,0 +1,224 @@
+
+For example, the following job installs the `SqlServer` and `PSScriptAnalyzer` modules:
+
+{% raw %}
+```yaml
+jobs:
+  install-dependencies:
+    name: Install dependencies
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install from PSGallery
+      shell: pwsh
+      run: |
+        Set-PSRepository PSGallery -InstallationPolicy Trusted
+        Install-Module SqlServer, PSScriptAnalyzer
+```
+{% endraw %}
+
+{% note %}
+
+**Note:** By default, no repositories are trusted by PowerShell. When installing modules from the PowerShell Gallery, you must explicitly set the installation policy for `PSGallery` to `Trusted`.
+
+{% endnote %}
+
+#### Caching dependencies
+
+You can cache PowerShell dependencies using a unique key, which allows you to restore the dependencies for future workflows with the [`cache`](https://github.com/marketplace/actions/cache) action. For more information, see "[Caching dependencies to speed up workflows](/actions/automating-your-workflow-with-github-actions/caching-dependencies-to-speed-up-workflows)."
+
+PowerShell caches its dependencies in different locations, depending on the runner's operating system. For example, the `path` location used in the following Ubuntu example will be different for a Windows operating system.
+
+{% raw %}
+```yaml
+steps:
+  - uses: actions/checkout@v2
+  - name: Setup PowerShell module cache
+    id: cacher
+    uses: actions/cache@v2
+    with:
+      path: "~/.local/share/powershell/Modules"
+      key: ${{ runner.os }}-SqlServer-PSScriptAnalyzer
+  - name: Install required PowerShell modules
+    if: steps.cacher.outputs.cache-hit != 'true'
+    shell: pwsh
+    run: |
+      Set-PSRepository PSGallery -InstallationPolicy Trusted
+      Install-Module SqlServer, PSScriptAnalyzer -ErrorAction Stop
+```
+{% endraw %}
+
+### Testing your code
+
+You can use the same commands that you use locally to build and test your code.
+
+#### Using PSScriptAnalyzer to lint code
+
+The following example installs `PSScriptAnalyzer` and uses it to lint all `ps1` files in the repository. For more information, see [PSScriptAnalyzer on GitHub](https://github.com/PowerShell/PSScriptAnalyzer).
+
+{% raw %}
+```yaml
+  lint-with-PSScriptAnalyzer:
+    name: Install and run PSScriptAnalyzer
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install PSScriptAnalyzer module
+      shell: pwsh
+      run: |
+            Set-PSRepository PSGallery -InstallationPolicy Trusted
+            Install-Module PSScriptAnalyzer -ErrorAction Stop
+    - name: Lint with PSScriptAnalyzer
+      shell: pwsh
+      run: |
+            Invoke-ScriptAnalyzer -Path *.ps1 -Recurse -Outvariable issues
+            $errors   = $issues.Where({$_.Severity -eq 'Error'})
+            $warnings = $issues.Where({$_.Severity -eq 'Warning'})
+            Write-Output "There were $($errors.Count) errors and $($warnings.Count) warnings total."
```

Comment:
@potatoqualitee 
What do you think of using `write-error` or `throw` so that failures/warnings will cause the job to fail? Maybe something like this?
```suggestion
            If($warnings.Count -gt '0' -or $($errors.Count) -gt '0'){write-error "There were $($errors.Count) errors and $($warnings.Count) warnings total."}

```

Reply:
Mine actually throw, too, not a bad idea. The only issue I have is with the syntax and capitalization of the proposed code. I'm also unsure if Write-Error throws an exception. To be sure, I'd add `-ErrorAction Stop`. I can't add a suggestion here, so I'll do it in the Files tab, brb

Reply:
I couldn't figure out how to suggest, so I added the following

````
if ($errors) {
    Write-Error "There were $($errors.Count) errors and $($warnings.Count) warnings total." -ErrorAction Stop
} else {
    Write-Output "There were $($errors.Count) errors and $($warnings.Count) warnings total."
}
````

Please let me know what you think. Warnings are common and can be looked over, so I just threw on error.

Reply:
@potatoqualitee - thanks so much, this looks good to me! 

Would it be useful to add the `ParseError` severity here? I saw that result come up in some of my Pester testing.

Reply:
@martin389 - I don't know what a `ParseError` severity is, sorry. I don't use PSScriptAnalyzer outside of tests that people put in our repo 😅

Reply:
No worries! I wasn't able to replicate it again in my testing, so it might have been a once-off event 😆